pkg/envoy/ads/cache_stream.go

```diff
@@ -8,46 +8,8 @@ import (
 	"github.com/envoyproxy/go-control-plane/pkg/cache/v3"
 
 	"github.com/openservicemesh/osm/pkg/envoy"
-	"github.com/openservicemesh/osm/pkg/messaging"
 )
 
-// Routine which fulfills listening to proxy broadcasts
-func (s *Server) watchForUpdates(ctx context.Context) {
-	// Register for proxy config updates broadcasted by the message broker
-	proxyUpdatePubSub := s.msgBroker.GetProxyUpdatePubSub()
-	proxyUpdateChan := proxyUpdatePubSub.Sub(messaging.ProxyUpdateTopic)
-	defer s.msgBroker.Unsub(proxyUpdatePubSub, proxyUpdateChan)
-
-	for {
-		select {
-		case <-proxyUpdateChan:
-			s.allPodUpdater()
-			// TODO(2683): listen to specific pod updates and cert rotations
-		case <-ctx.Done():
-			return
-		}
-	}
-}
-
-func (s *Server) allPodUpdater() {
```

Comment:
We really updated all the pods every time a single pod got an update event? Big save of resources here!